lib/ssl/test/ssl_ocsp_SUITE.erl

```diff
@@ -0,0 +1,367 @@
+%%
+%% %CopyrightBegin%
+%%
+%% Copyright Ericsson AB 2011-2020. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% %CopyrightEnd%
+%%
+
+-module(ssl_ocsp_SUITE).
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("public_key/include/public_key.hrl").
+
+%% Note: This directive should only be used in test suites.
+-compile(export_all).
+
+%%--------------------------------------------------------------------
+%% Common Test interface functions -----------------------------------
+%%--------------------------------------------------------------------
+all() -> 
+    [{group, tls1_2},
+     {group, tls1_3},
+     {group, revoked_1_2},
+     {group, revoked_1_3}].
+
+groups() -> 
+    [{tls1_2, [], tls1_2_tests()},
+     {tls1_3, [], tls1_3_tests()},
+     {revoked_1_2, [], tls1_2_revoked_tests()},
+     {revoked_1_3, [], tls1_3_revoked_tests()}].
+
+tls1_2_tests() ->
+    [ocsp_stapling_without_nonce_and_responder_certs_tls1_2,
+     ocsp_stapling_with_nonce_tls1_2,
+     ocsp_stapling_with_responder_cert_tls1_2].
+
+tls1_2_revoked_tests() ->
+    [ocsp_stapling_revoked_tls1_2].
+
+tls1_3_tests() ->
+    [ocsp_stapling_without_nonce_and_responder_certs_tls1_3,
+     ocsp_stapling_with_nonce_and_responder_certs_tls1_3].
+
+tls1_3_revoked_tests() ->
+    [ocsp_stapling_revoked_tls1_3].
+
+%%--------------------------------------------------------------------
+init_per_suite(Config) ->
+    catch crypto:stop(),
+    try crypto:start() of
+	ok ->
+        ssl_test_lib:clean_start(),
+        DataDir = proplists:get_value(data_dir, Config),
+        PrivDir = proplists:get_value(priv_dir, Config),
+
+        %% Prepare certs
+        {ok, _} = make_certs:all(DataDir, PrivDir),
```

Comment:
We should only generate one cert type.

Reply:
I set this to resolved as this is the only way today to generate revoked certificates.